CRM/Import/DataSource.php

```diff
@@ -115,31 +177,64 @@ protected function getSubmittedValue(string $valueName) {
    *
    * @param int $limit
    * @param int $offset
+   * @param array $statuses
+   * @param bool $nonAssociative
+   *   Return as a non-associative array?
    *
    * @return array
    *
    * @throws \API_Exception
    * @throws \CRM_Core_Exception
    */
-  public function getRows(int $limit = 0, int $offset = 0) {
-    $query = 'SELECT * FROM ' . $this->getTableName();
-    if ($limit) {
-      $query .= ' LIMIT ' . $limit . ($offset ? (' OFFSET ' . $offset) : NULL);
-    }
+  public function getRows(bool $nonAssociative = TRUE): array {
```

Comment:
Function was only just added so we can change the signature - only one call to it now - https://github.com/civicrm/civicrm-core/pull/23263/files#diff-4f0a1c0d1f7a02c448ebe273bd5b951a1811dd14949b11b9c8914b2c198fcebaR384